basic/07a_robot_3_dof_arm.cpp

```diff
@@ -0,0 +1,175 @@
+/**
+ * Put everything together to control a 3-DoF arm.
+ *
+ * For more information, go to http://docs.hebi.us/tools.html#cpp-api
+ *
+ * This script assumes you can create a group with 1 module.
+ *
+ * HEBI Robotics
+ * October 2018
+ */
+
+#include <iostream>
+#include <chrono>
+#include <thread>
+
+#include "lookup.hpp"
+#include "group_command.hpp"
+#include "group_feedback.hpp"
+#include "robot_model.hpp"
+#include "trajectory.hpp"
+
+// The examples provide this utility function for computing gravity compensation
+// efforts.
+#include "util/grav_comp.hpp"
+
+#ifndef M_PI
+#define M_PI 3.14159265358979323846
+#endif
+
+using namespace hebi;
+
+/// A helper function to create a group from named modules, and set specified
+/// gains on the modules in that group.
+std::shared_ptr<Group> getGroup() {
+  // Get group
+  std::vector<std::string> families {"3-DoF Arm"};
```

Comment:
You didn't include `<string>` **or** `<vector>`

Reply:
Not directly, but they are included in our headers.  I also didn't include `<memory>`.  I had always assumed it was reasonable practice not to include every single header necessary if they are covered in your other headers.

Reply:
Yeah you're right, I'm sorry. See my comment above about `lookup.hpp`